main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidatorTest.java

```diff
@@ -17,110 +17,356 @@
+            .setStopId("stop id 2")
+            .setStopSequence(2)
+            .setTimepoint(1)
+            .build());
+    assertThat(generateNotices(createHeaderWithTimepointColumn(), stopTimes))
         .containsExactly(
-            new StopTimeTimepointWithoutTimesNotice(
-                1, "first trip id", 2, GtfsStopTimeTableLoader.ARRIVAL_TIME_FIELD_NAME),
-            new StopTimeTimepointWithoutTimesNotice(
-                1, "first trip id", 2, GtfsStopTimeTableLoader.DEPARTURE_TIME_FIELD_NAME));
+            new StopTimeTimepointWithoutTimesNotice(stopTimes.get(0), ARRIVAL_TIME_FIELD_NAME),
+            new StopTimeTimepointWithoutTimesNotice(stopTimes.get(0), DEPARTURE_TIME_FIELD_NAME));
   }
 
   @Test
   public void timepointWithBothTimesShouldNotGenerateNotice() {
-    assertThat(
-            generateNotices(
-                new GtfsStopTime.Builder()
-                    .setCsvRowNumber(1)
-                    .setTripId("first trip id")
-                    .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
-                    .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
-                    .setStopId("stop id")
-                    .setStopSequence(2)
-                    .setTimepoint(1)
-                    .build()))
-        .isEmpty();
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("first trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id")
+            .setStopSequence(2)
+            .setTimepoint(1)
+            .build());
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("second trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id 2")
+            .setStopSequence(2)
+            .setTimepoint(1)
+            .build());
+    assertThat(generateNotices(createHeaderWithTimepointColumn(), stopTimes)).isEmpty();
   }
 
   @Test
-  public void missingDepartureTimeShouldGenerateNotice() {
-    assertThat(
-            generateNotices(
-                new GtfsStopTime.Builder()
-                    .setCsvRowNumber(1)
-                    .setTripId("first trip id")
-                    .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
-                    .setDepartureTime(null)
-                    .setStopId("stop id")
-                    .setStopSequence(2)
-                    .setTimepoint(1)
-                    .build()))
+  public void timepoint_missingDepartureTimeShouldGenerateNotice() {
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(
+        new GtfsStopTime.Builder()
```

Comment:
Do you need both stopTime records for this test? If not for simplicity I'd remove the unneeded code. Same for other tests.

Reply:
It is actually not needed. Removed in https://github.com/MobilityData/gtfs-validator/pull/1044/commits/e33c960a486db0db6d2af6a0b705fcfd6b44b026.